addons/account/models/account_move.py

```diff
@@ -377,118 +1907,123 @@ def button_cancel(self):
+            .with_context(active_ids=self.ids, active_model='account.move', active_id=self.id)\
+            .action_register_payment()
+
+    @api.multi
+    def _get_report_base_filename(self):
+        return self._get_invoice_display_name()
 
     @api.multi
-    def _reverse_move(self, date=None, journal_id=None):
+    def preview_invoice(self):
         self.ensure_one()
-        date = date or fields.Date.today()
-        with self.env.norecompute():
-            reversed_move = self.copy(default={
-                'date': date,
-                'journal_id': journal_id.id if journal_id else self.journal_id.id,
-                'ref': _('Reversal of: %s') % (self.name),
-            })
-            for acm_line in reversed_move.line_ids.with_context(check_move_validity=False):
-                acm_line.write({
-                    'debit': acm_line.credit,
-                    'credit': acm_line.debit,
-                    'amount_currency': -acm_line.amount_currency
-                })
-            self.reverse_entry_id = reversed_move
-        self.recompute()
-        return reversed_move
-
-    @api.multi
-    def reverse_moves(self, date=None, journal_id=None):
-        date = date or fields.Date.today()
-        reversed_moves = self.env['account.move']
-        for ac_move in self:
-            #unreconcile all lines reversed
-            aml = ac_move.line_ids.filtered(lambda x: x.account_id.reconcile or x.account_id.internal_type == 'liquidity')
-            aml.remove_move_reconcile()
-            reversed_move = ac_move._reverse_move(date=date, journal_id=journal_id)
-            reversed_moves |= reversed_move
-            #reconcile together the reconcilable (or the liquidity aml) and their newly created counterpart
-            for account in set([x.account_id for x in aml]):
-                to_rec = aml.filtered(lambda y: y.account_id == account)
-                to_rec |= reversed_move.line_ids.filtered(lambda y: y.account_id == account)
-                #reconciliation will be full, so speed up the computation by using skip_full_reconcile_check in the context
-                to_rec.reconcile()
-        if reversed_moves:
-            reversed_moves._post_validate()
-            reversed_moves.post()
-            return [x.id for x in reversed_moves]
-        return []
+        return {
+            'type': 'ir.actions.act_url',
+            'target': 'self',
+            'url': self.get_portal_url(),
+        }
 
     @api.multi
-    def open_reconcile_view(self):
-        return self.line_ids.open_reconcile_view()
+    def _compute_access_url(self):
+        super(AccountMove, self)._compute_access_url()
+        for move in self.filtered(lambda move: move.type in ('out_invoice', 'out_refund', 'in_invoice', 'in_refund')):
+            move.access_url = '/my/invoices/%s' % (move.id)
 
-    # FIXME: Clarify me and change me in master
     @api.multi
-    def action_duplicate(self):
```

Comment:
why removing this?

There is a new button on the account.move.line grouped list view that tries to use it